### PR TITLE
`esp-wifi` docs improve

### PR DESCRIPTION
**Pull request by playfulFence**
### Submission Checklist 📝
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

#### Description
closes https://github.com/esp-rs/esp-hal/issues/2296

#### Testing
Built docs and checked if everything looks correct

### CODE REVIEW

File: esp-wifi/src/wifi/mod.rs [hunk 1]
```diff
@@ -142,75 +142,141 @@ use crate::{
     compat::queue::SimpleQueue,
 };
 
+/// Enumeration of supported Wi-Fi authentication methods.
```
Comment (bugadani) [inline esp-wifi/src/wifi/mod.rs:145]:
```suggestion
/// Supported Wi-Fi authentication methods.
```

File: esp-wifi/src/wifi/mod.rs [hunk 2]
```diff
@@ -142,75 +142,141 @@ use crate::{
...
     WPA3Personal,
+
+    /// WPA2/WPA3 Personal authentication (supports both).
     WPA2WPA3Personal,
+
+    /// WLAN Authentication and Privacy Infrastructure (WAPI).
     WAPIPersonal,
 }
 
+/// Enumeration of supported Wi-Fi protocols.
```
Comment (bugadani) [inline esp-wifi/src/wifi/mod.rs:180]:
```suggestion
/// Supported Wi-Fi protocols.
```

File: esp-wifi/src/wifi/mod.rs [hunk 3]
```diff
@@ -142,75 +142,141 @@ use crate::{
...
     P802D11BGNLR,
+
+    /// 802.11 long-range (LR) protocol.
     P802D11LR,
+
+    /// 802.11b/g/n/ax protocol.
     P802D11BGNAX,
 }
 
+/// Enumeration of secondary Wi-Fi channels.
```
Comment (bugadani) [inline esp-wifi/src/wifi/mod.rs:206]:
```suggestion
/// Secondary Wi-Fi channels.
```

File: esp-wifi/src/esp_now/mod.rs [hunk 4]
```diff
@@ -71,31 +90,39 @@ impl Error {
...
     }
 }
 
+/// Common errors that can occur while using ESP-NOW driver.
 #[derive(Debug)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub enum EspNowError {
+    /// Internal Error.
     Error(Error),
+    /// Sending an ESP-NOW message failed.
```
Comment (bugadani) [inline esp-wifi/src/esp_now/mod.rs:108]:
```suggestion
    /// Failed to send an ESP-NOW message.
```

File: esp-wifi/src/esp_now/mod.rs [hunk 5]
```diff
@@ -169,33 +196,57 @@ pub enum WifiPhyRate {
...
+    /// softap.
     pub peer_address: [u8; 6],
+
+    /// ESP-NOW peer local master key that is used to encrypt data.
     pub lmk: Option<[u8; 16]>,
+
+    /// Wi-Fi channel that peer uses to send/receive ESP-NOW data.
     pub channel: Option<u8>,
+
+    /// ESP-NOW data that this peer sends/receives is encrypted or not.
```
Comment (bugadani) [inline esp-wifi/src/esp_now/mod.rs:213]:
```suggestion
    /// Whether the data sent/received by this peer is encrypted.
```

File: esp-wifi/src/esp_now/mod.rs [hunk 6]
```diff
@@ -169,33 +196,57 @@ pub enum WifiPhyRate {
...
+
+    /// The buffer containing the received data.
     pub data: [u8; 256],
+
+    /// Information about a received packet.
     pub info: ReceiveInfo,
 }
 
 impl ReceivedData {
+    /// Returns a slice of the received data, limited to the actual length of
```
Comment (bugadani) [inline esp-wifi/src/esp_now/mod.rs:248]:
```suggestion
    /// Returns the received payload.
```

File: esp-wifi/src/esp_now/mod.rs [hunk 7]
```diff
@@ -478,13 +532,14 @@ impl<'s> Drop for SendWaiter<'s> {
     }
 }
 
-/// This is the sender part of ESP-NOW. You can get this sender by splitting
+/// This is the receiver part of ESP-NOW. You can get this receiver by splitting
```
Comment (bugadani) [inline esp-wifi/src/esp_now/mod.rs:535]:
```suggestion
/// This is the receiver part of ESP-NOW. You can get this receiver by splitting.
```

File: esp-wifi/src/esp_now/mod.rs [hunk 8]
```diff
@@ -630,6 +687,8 @@ impl<'d> EspNow<'d> {
         Ok(esp_now)
     }
 
+    /// Splits the ESP-NOW instance into its manager, sender, and receiver
```
Comment (bugadani) [inline esp-wifi/src/esp_now/mod.rs:690]:
```suggestion
    /// Splits the `EspNow` instance into its manager, sender, and receiver
```

I guess we should be consistent how we refer to the technology vs the implementation.

File: esp-wifi/src/esp_now/mod.rs [hunk 9]
```diff
@@ -685,35 +744,35 @@ impl<'d> EspNow<'d> {
...
+    /// Set wake window for esp_now to wake up in interval unit.
     ///
     /// Window is milliseconds the chip keep waked each interval, from 0 to
     /// 65535.
     pub fn set_wake_window(&self, wake_window: u16) -> Result<(), EspNowError> {
         self.manager.set_wake_window(wake_window)
     }
 
-    /// Config ESPNOW rate
+    /// Config ESP-NOW rate.
```
Comment (bugadani) [inline esp-wifi/src/esp_now/mod.rs:770]: This needs some more attention I think.

File: esp-wifi/src/tasks.rs [hunk 10]
```diff
@@ -13,6 +14,8 @@ pub fn init_tasks() {
     task_create(timer_task, core::ptr::null_mut(), 8192);
 }
 
+/// Entry point for the timer task responsible for handling scheduled timer
+/// events.
 pub extern "C" fn timer_task(_param: *mut esp_wifi_sys::c_types::c_void) {
```
Comment (bugadani) [inline esp-wifi/src/tasks.rs:19]: Does this have to be public?

File: esp-wifi/src/wifi/mod.rs [hunk 11]
```diff
@@ -451,6 +598,8 @@ impl Configuration {
         }
     }
 
+    /// Retrieves mutable references to both the `ClientConfiguration`
+    /// and `AccessPointConfiguration` contained within a `Configuration`.
```
Comment (bugadani) [inline esp-wifi/src/wifi/mod.rs:602]:
```suggestion
    /// Retrieves mutable references to both the `ClientConfiguration`
    /// and `AccessPointConfiguration`.
```

File: esp-wifi/src/wifi/mod.rs [hunk 12]
```diff
@@ -499,6 +650,7 @@ pub mod ipv4 {
     impl FromStr for Mask {
         type Err = &'static str;
 
+        /// Parses a string to create a `Mask`.
```
Comment (bugadani) [inline esp-wifi/src/wifi/mod.rs:653]: I don't think this or any other trait impl needs to be documented. The traits are pretty obvious and have their own docs. There are quite a few of these.

File: esp-wifi/src/wifi/mod.rs [hunk 13]
```diff
@@ -614,29 +782,44 @@ pub mod ipv4 {
...
     pub enum ClientConfiguration {
+        /// Use DHCP to obtain network settings.
         DHCP(DHCPClientSettings),
+
+        /// Use a fixed configuration for network settings.
         Fixed(ClientSettings),
     }
 
     impl ClientConfiguration {
+        /// Returns a reference to the fixed settings if the client is using a
```
Comment (bugadani) [inline esp-wifi/src/wifi/mod.rs:812]: Smells like ChatGPT :)

File: esp-wifi/src/wifi/mod.rs [hunk 14]
```diff
@@ -1868,37 +2136,46 @@ mod sealed {
 
 use sealed::*;
 
+/// Provides methods for retrieving the Wi-Fi mode and MAC address.
 pub trait WifiDeviceMode: Sealed {
+    /// Returns the Wi-Fi mode (e.g., Station, Access Point).
```
Comment (bugadani) [inline esp-wifi/src/wifi/mod.rs:2141]:
```suggestion
    /// Returns the currently active Wi-Fi mode.
```

File: esp-wifi/src/wifi/mod.rs [hunk 15]
```diff
@@ -1868,37 +2136,46 @@ mod sealed {
...
     fn mac_address(self) -> [u8; 6];
 }
 
+/// Wi-Fi station device.
 #[derive(Debug, Clone, Copy, PartialEq)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub struct WifiStaDevice;
 
 impl WifiDeviceMode for WifiStaDevice {
+    /// Returns the mode of the Wi-Fi device as `Sta` (Station).
```
Comment (bugadani) [inline esp-wifi/src/wifi/mod.rs:2154]:
```suggestion
```

File: esp-wifi/src/wifi/mod.rs [hunk 16]
```diff
@@ -1868,37 +2136,46 @@ mod sealed {
...
     }
 }
 
+/// Wi-Fi Access Point (AP) device.
 #[derive(Debug, Clone, Copy, PartialEq)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub struct WifiApDevice;
 
 impl WifiDeviceMode for WifiApDevice {
+    /// Returns the mode of the Wi-Fi device as `Ap` (Access Point).
```
Comment (bugadani) [inline esp-wifi/src/wifi/mod.rs:2173]:
```suggestion
```

File: esp-wifi/src/tasks.rs [hunk 17]
```diff
@@ -5,6 +5,7 @@ use crate::{
     timer::{get_systimer_count, yield_task},
 };
 
+/// Initializes the `main` and `timer` tasks for the Wi-Fi driver.
 pub fn init_tasks() {
```
Comment (MabezDev) [inline esp-wifi/src/tasks.rs:9]: I think this can be `pub(crate)`?

Comment (playfulFence) [reply esp-wifi/src/tasks.rs:9]: I guess the same as : https://github.com/esp-rs/esp-hal/pull/2354#discussion_r1802610767

Well, I agree, also it doesn't go out far beyond the initial task, so I'll probably make those functions be `pub(crate)` , yes.

File: esp-wifi/src/esp_now/mod.rs (hunk 7 above)
Comment (bugadani) [reply esp-wifi/src/esp_now/mod.rs:534]: duh I didn't notice that the comment continues

**Review by jessebraham — APPROVED** [review]
Thanks, LGTM!

**Review by bjoernQ — APPROVED** [review]
Thanks

**Pull request merged**